src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseAssignableTypeForForeachVariableTests.cs

```diff
@@ -0,0 +1,466 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.CSharp.Analyzers.Runtime.CSharpUseAssignableTypeForForeachVariable,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class UseAssignableTypeForForeachVariableTests
+    {
+        [Fact]
+        public async Task SameType()
+        {
+            var test = @"
+using System.Collections.Generic;
+
+namespace ConsoleApplication1
+{
+    class Program
+    {   
+        void Main()
+        {
+            var x = new List<string>();
+            foreach (string item in x)
+            {
+            }
+        }
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(test);
+        }
+
+        [Fact]
+        public async Task CastBaseToChild()
+        {
+            var test = @"
+using System.Collections.Generic;
+
+namespace ConsoleApplication1
+{
+    class Program
+    {   
+        void Main()
+        {
+            var x = new List<A>();
+            foreach (B item in x)
```

Comment:
As you have some arguments to provide you need to use the markup syntax with an id

(note that I am not sure where you report the issue so you will have to tweak the location)
```suggestion
            {|#0:foreach (B item in x)|}
```